### PR TITLE
ci: Fix change detection (part 2)

### PR DESCRIPTION
**Pull request by phlax**
- fix stage dependency
- fix change detection
 - add some debugging information
    
Signed-off-by: Ryan Northey <ryan@synca.io>

---

This fixes the `jq` expression to use "slurp" to ensure the json is a single list (and works with subsequent code), and adds some debugging info

also adds a `not(cancelled())` condition to the `check` stage as the dep on `env` is triggering this when the run is cancelled

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by phlax** [comment]
@alyssawilk apologies - i thought i had tested this bit out sufficiently - but evidently not

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -274,7 +279,7 @@ stages:
         AZP_BRANCH: $(Build.SourceBranch)
 
 - stage: linux_x64
-  dependsOn: ["precheck"]
+  dependsOn: ["env", "precheck"]
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:282]: rm

**Comment by phlax** [comment]
this seems to work:

https://dev.azure.com/cncf/envoy/_build/results?buildId=122158&view=results

File: .azure-pipelines/pipelines.yml [hunk 2]
```diff
@@ -374,11 +379,12 @@ stages:
 
 - stage: check
   dependsOn: ["env", "linux_x64"]
-  condition: or(eq(variables['PostSubmit'], true), and(ne(dependencies.env.repo.outputs['changed.mobileOnly'], 'true'), ne(dependencies.env.repo.outputs['changed.docsOnly'], 'true')))
   jobs:
   - job: bazel
     displayName: "linux_x64"
     dependsOn: []
+    # Skip checks if only mobile/ or docs/ have changed.
+    condition: and(not(canceled()), succeeded(), ne(stageDependencies.env.repo.outputs['changed.mobileOnly'], 'true'), ne(stageDependencies.env.repo.outputs['changed.docsOnly'], 'true'))
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:387]: this doesnt currently check for `PostSubmit` but im thinking that conditions are already a bit too complicated so probably would be better to hack where `mobileOnly` etc is set and make that conditional on not `PostSubmit`

Comment (phlax) [reply .azure-pipelines/pipelines.yml:394]: i have updated the `env` stage

File: .azure-pipelines/pipelines.yml [hunk 3]
```diff
@@ -35,15 +35,18 @@ stages:
   #
   # To use a variable as a condition for a `stage` you can specify it as follows:
   #
-  # condition: ne(dependencies.env.outputs['repo.changed.mobileOnly'], 'true')
+  # condition: and(not(canceled()), succeeded(), ne(dependencies.env.repo.outputs['changed.mobileOnly'], 'true'))
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:38]: revert change to `.repo.`

**Review by adisuissa — APPROVED** [review]
LGTM, thanks!

**Review by RyanTheOptimist — APPROVED** [review]
Fingers crossed!

**Pull request merged**